packages/editor/src/store/selectors.js

```diff
@@ -130,6 +130,22 @@ export const hasNonPostEntityChanges = createRegistrySelector(
 	}
 );
 
+/**
+ * Returns true if there are unsaved changes to the
+ * post's meta fields, and false otherwise.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {boolean} Whether there are edits or not.
+ */
+export const hasMetaChanges = createRegistrySelector( ( select ) => () => {
+	const dirtyEntityRecords =
+		select( coreStore ).__experimentalGetDirtyEntityRecords();
+	return dirtyEntityRecords.some(
+		( entityRecord ) => entityRecord.hasMetaChanges
+	);
+} );
```

Comment:
As it is still not 100% clear if we will need this specific selector, or we might want to abstract it to not only track meta changes as suggested [here](https://github.com/WordPress/gutenberg/pull/61811#discussion_r1615603578), maybe it is a good idea to make it private for now?

Reply:
I don't have strong opinions about whether there would be a new selector introduced given that there is `hasNonPostEntityChanges`, which looks very similar. It would definitely help to make it private for the start as it needs more testing (unit tests would help here, too). The function will return `true` if any entity has changed. So it would be valuable to at least narrow down the check to `entityRecord.kind === 'postType'`. In effect, the name would be `hasPostMetaChanges`. It's worth also raising the question, whether we want to observe change to all post types or it should be narrowed down to the currently edited post id.

Reply:
I was the one suggesting creating a new selector or renaming `hasNonPostEntityChanges`: [link](https://github.com/WordPress/gutenberg/pull/61811#discussion_r1615606135). For me, it felt a bit weird because it actually has `postEntityChanges` when the meta fields change. Having said that, I don't have a strong opinion either.

> So it would be valuable to at least narrow down the check to entityRecord.kind === 'postType'.

This makes sense to me.

> whether we want to observe change to all post types or it should be narrowed down to the currently edited post id.

If we do this, we have to ensure it works properly with the query loop, which includes multiple posts that can be edited.